### ISSUE TITLE
[Suggestion] Show processing indicator

### ISSUE THREAD

**Issue by etlweather** [theJayTea/WritingTools#28]
It can sometime take a bit of time before the LLM finishes. Showing a "processing" indicator would tell me it is doing something. Right now it first looks like nothing happened.

**Comment by etlweather** [comment]
(Nice little app by the way!)

**Comment by theJayTea** [comment]
Ah! While I mostly use it with the super quick Gemini 1.5 Flash with shorter text, I understand that it can take a moment with longer text, especially with local models.
Do you have any ideas for UX that would convey this, while still staying out of the user's way?
Maybe converting the cursor to the loading icon?

**Comment by theJayTea** [comment]
@momokrono , thanks so much for adding that! :)
I'll join in to try and help make this pretty (and also work on the other issues and features) after Nov 1st :)

**Comment by aimagist** [comment]
The taskbar icon can signal this processing as well, since it's already there, it wouldn't spam the UI/UX at all

**Comment by EmilPetrakiev** [comment]
Why don't you just stream the input from the LLM and if the user exits the targeted window or presses the ESC key just stop the generation.

**Comment by momokrono** [comment]
@EmilPetrakiev You mean like streaming? This can be enabled already via config.json.
Not showing some sort of status-indicator could be a good idea when using ChatGPT or Gemini, when using a local inference server like ollama if you have to re-load the model back into VRAM you will be waiting quite a lot, especially for large models. Not only that, you would be waiting for the context to be processed too before you receive the first chunk of the response. In this case a progress bar or a spinning icon would tell the user "just wait, something is going on in the background but everything is fine".
If I misunderstood what you meant, please correct me.

**Comment by EmilPetrakiev** [comment]
@momokrono I did not know that streaming could be enabled in the config.json. But yea, my idea was that it should have a streamed output so the user can start using the data that the LLM provides while it's still generating it. I forgot to think about the time when the model still loads. It is a great idea to have for example the cursor of the mouse become the busy cursor (The spinning one) like someone above suggested.

**Comment by momokrono** [comment]
@EmilPetrakiev no worries!

Yes, a spinning cursor could be a good idea, but ideally it should be used only if the mouse cursor is hovering over the response window IMHO. For example, if the user moves the mouse over a different window the cursor should go back to being "normal" to allow for usual interaction, or it would pass the idea of the whole sistem loading something. But I'm no UX designer so maybe this is what we want instead?

In any case, something smaller than a loading bar would definitely feel more polished.

@theJayTea what do you think?

**Comment by theJayTea** [comment]
Hello everyone!

I think a global busy cursor would be best. The app window can disappear as it already does, and the cursor gets into spinny mode until the text is replaced. I feel this would be the most unobtrusive UX as
- the pop-up window would still get out of the way as soon as you click, making it feel light and instant
- the global cursor importantly signals to the user that they can't click away from the textbox for these few seconds and go somewhere else, as that would ruin the pasting process.

PS: @momokrono had tried to implement a loading indicator (and not the one described right above) in the `dev` branch earlier, but when I tested it on Windows it sadly didn't work.

I think the above approach would be best, and I could try implementing it when I get time.

**Comment by WYang555** [comment]
Yes, all of these are brilliant ideas; I believe that if the window can remain while I evoke the application, it will enhance the experience.
Now I have to evoke the window every time the application processes a question.

**Comment by momokrono** [comment]
The feature has been present for quite a while now, closing!
